src/estimation/src/BerdyHelper.cpp

```diff
@@ -410,28 +445,77 @@ TraversalIndex getTraversalIndexFromJointIndex(const Model & m_model, const Trav
 
 IndexRange BerdyHelper::getRangeJointVariable(BerdyDynamicVariablesTypes dynamicVariableType, JointIndex idx)
 {
-    assert(m_options.berdyVariant == ORIGINAL_BERDY_FIXED_BASE);
-
     if( !isJointBerdyDynamicVariable(dynamicVariableType) )
     {
         assert(false);
         return IndexRange::InvalidRange();
     }
 
-    return getRangeOriginalBerdyFixedBase(dynamicVariableType,getTraversalIndexFromJointIndex(m_model,m_dynamicsTraversal,idx));
+    if (m_options.berdyVariant == ORIGINAL_BERDY_FIXED_BASE)
+    {
+        return getRangeOriginalBerdyFixedBase(dynamicVariableType,
+                                              getTraversalIndexFromJointIndex(m_model, m_dynamicsTraversal, idx));
+    }
+    else
+    {
+        IndexRange ret;
+        assert(m_options.berdyVariant == BERDY_FLOATING_BASE);
+        assert(m_options.includeAllNetExternalWrenchesAsDynamicVariables);
+
+        int totalSizeOfLinkVariables  = 12*m_model.getNrOfLinks();
+
+        switch (dynamicVariableType)
+        {
+            case JOINT_WRENCH:
+                ret.offset = totalSizeOfLinkVariables + 6*idx;
+                ret.size = 6;
+                break;
+            default:
+                assert(false);
+                ret = IndexRange::InvalidRange();
+                break;
+        }
+
+        return ret;
+    }
 }
 
 IndexRange BerdyHelper::getRangeDOFVariable(BerdyDynamicVariablesTypes dynamicVariableType, DOFIndex idx)
 {
-    assert(m_options.berdyVariant == ORIGINAL_BERDY_FIXED_BASE);
-
     if( !isDOFBerdyDynamicVariable(dynamicVariableType) )
     {
         return IndexRange::InvalidRange();
     }
 
-    // For a model with all 1-dofs, dof index and joint index matches
-    return getRangeOriginalBerdyFixedBase(dynamicVariableType,getTraversalIndexFromJointIndex(m_model,m_dynamicsTraversal,idx));
+    if (m_options.berdyVariant == ORIGINAL_BERDY_FIXED_BASE)
+    {
+        // For a model with all 1-dofs, dof index and joint index matches
+        return getRangeOriginalBerdyFixedBase(dynamicVariableType,
+                                              getTraversalIndexFromJointIndex(m_model, m_dynamicsTraversal, idx));
+    }
+    else
+    {
+        IndexRange ret;
+        assert(m_options.berdyVariant == BERDY_FLOATING_BASE);
+        assert(m_options.includeAllNetExternalWrenchesAsDynamicVariables);
+
+        int totalSizeOfLinkVariables  = 12*m_model.getNrOfLinks();
```

Comment:
Just a comment: probably with a C++ enum class we can group all the information regarding the variables in a single class, e.g. size of all link variables = 12

Reply:
or.. I was thinking to something like:
```cpp
class BerdyDynamicsVariables
{
public:
    enum _hidden_enum {
       enum_1 = value,
       enum_2 = value2, ...
   };

private: 
   std::set< _hidden_enum> m_enumValues; // (so we can provide an iterator on the enum values)

public:
   // other public methods, like
  size_t totalSize() { return 12; }
 ...
};
```

Reply:
Unfortunatly that `12` depends on the berdy variant used at this moment.